io/src/real_sense_2_grabber.cpp

```diff
@@ -0,0 +1,349 @@
+#endif
+    for (int index = 0; index < cloud->points.size (); ++index)
+    {
+      auto ptr = vertices_ptr + index;
+      auto uvptr = texture_ptr + index;
+      auto& p = cloud->points[index];
+
+      p.x = ptr->x;
+      p.y = ptr->y;
+      p.z = ptr->z;
+
+      std::tie (r, g, b) = getTextureColor (ir, uvptr->u, uvptr->v);
+
+      p.intensity = 0.299f * static_cast <float> (r) + 0.587f * static_cast <float> (g) + 0.114f * static_cast <float> (b);
+
+    }
+
+    return cloud;
+  }
+
+  pcl::PointCloud<pcl::PointXYZRGB>::Ptr 
+  RealSense2Grabber::convertRGBDepthToPointXYZRGB (const rs2::points & points, rs2::video_frame & rgb)
+  {
+    pcl::PointCloud<pcl::PointXYZRGB>::Ptr cloud (new pcl::PointCloud<pcl::PointXYZRGB> ());
+
+    auto sp = points.get_profile ().as<rs2::video_stream_profile> ();
+    cloud->width = sp.width ();
+    cloud->height = sp.height ();
+    cloud->is_dense = false;
+    cloud->points.resize (points.size ());
+
+    auto vertices_ptr = points.get_vertices ();
+    auto texture_ptr = points.get_texture_coordinates ();
+
+#ifdef _OPENMP
+#pragma omp parallel for 
+#endif
+    for (int index = 0; index < cloud->points.size (); ++index)
+    {
+      auto ptr = vertices_ptr + index;
+      auto uvptr = texture_ptr + index;
+      auto& p = cloud->points[index];
+
+      p.x = ptr->x;
+      p.y = ptr->y;
+      p.z = ptr->z;
+
+      std::tie (p.r, p.g, p.b) = getTextureColor (rgb, uvptr->u, uvptr->v);
+
+      p.a = 255;
+    }
+
+    return cloud;
+  }
+
+  pcl::PointCloud<pcl::PointXYZRGBA>::Ptr 
+  RealSense2Grabber::convertRGBADepthToPointXYZRGBA (const rs2::points & points, rs2::video_frame & rgb)
+  {
+    pcl::PointCloud<pcl::PointXYZRGBA>::Ptr cloud (new pcl::PointCloud<pcl::PointXYZRGBA> ());
+
+    auto sp = points.get_profile ().as<rs2::video_stream_profile> ();
+    cloud->width = sp.width ();
+    cloud->height = sp.height ();
+    cloud->is_dense = false;
+    cloud->points.resize (points.size ());
+
+    auto vertices_ptr = points.get_vertices ();
+    auto texture_ptr = points.get_texture_coordinates ();
+
+#ifdef _OPENMP
+#pragma omp parallel for 
+#endif
+    for (int index = 0; index < cloud->points.size (); ++index)
+    {
+      auto ptr = vertices_ptr + index;
+      auto uvptr = texture_ptr + index;
+      auto& p = cloud->points[index];
+
+      p.x = ptr->x;
+      p.y = ptr->y;
+      p.z = ptr->z;
+
+      std::tie (p.r, p.g, p.b) = getTextureColor (rgb, uvptr->u, uvptr->v);
+
+      p.a = 255;
+    }
+
+    return cloud;
+  }
+
+  std::tuple<uint8_t, uint8_t, uint8_t> 
+  RealSense2Grabber::getTextureColor (rs2::video_frame &texture, float u, float v)
+  {
+    auto ptr = dynamic_cast<rs2::video_frame*>(&texture);
+
+    const int w = ptr->get_width (), h = ptr->get_height ();
+    int x = std::min (std::max (int (u*w + .5f), 0), w - 1);
+    int y = std::min (std::max (int (v*h + .5f), 0), h - 1);
+    int idx = x * ptr->get_bits_per_pixel () / 8 + y * ptr->get_stride_in_bytes ();
```

Comment:
Is this all right?
```patch
- int idx = x * ptr->get_bits_per_pixel () / 8 + y * ptr->get_stride_in_bytes ();
+ int idx = x * ptr->get_bytes_per_pixel () + y * ptr->get_stride_in_bytes ();
```

Reply:
It could be, let me test it fast.  This is where I got the original code from to get rgb out of the texture:

https://github.com/IntelRealSense/librealsense/blob/2b8ff8ce95c15c6d6398fdee70303fc8d9aaa79a/src/archive.cpp#L27

Reply:
Still works!

![image](https://user-images.githubusercontent.com/3513250/36314632-491e06b0-12ea-11e8-8b65-05f5059e8e2c.png)